kore/src/Kore/Repl/Interpreter.hs

```diff
@@ -1321,31 +1325,32 @@ showRewriteRule rule =
     makeKoreReplOutput (unparseToString rule)
     <> makeAuxReplOutput (show . Pretty.pretty . from @_ @SourceLocation $ rule)
 
--- | Unparses a strategy node, using an omit list to hide specified children.
-unparseClaimStateComponent
+-- | Pretty prints a strategy node, using an omit list to hide specified children.
+prettyClaimStateComponent
     :: (SomeClaim -> Pattern RewritingVariableName)
     -> Set String
     -- ^ omit list
     -> CommonClaimState
     -- ^ pattern
     -> ReplOutput
-unparseClaimStateComponent transformation omitList =
+prettyClaimStateComponent transformation omitList =
     claimState ClaimStateTransformer
         { claimedTransformer =
-            makeKoreReplOutput . unparseComponent
+            makeKoreReplOutput . prettyComponent
         , remainingTransformer = \goal ->
             makeAuxReplOutput "Stuck: \n"
-            <> makeKoreReplOutput (unparseComponent goal)
+            <> makeKoreReplOutput (prettyComponent goal)
         , rewrittenTransformer =
-            makeKoreReplOutput . unparseComponent
+            makeKoreReplOutput . prettyComponent
         , stuckTransformer = \goal ->
             makeAuxReplOutput "Stuck: \n"
-            <> makeKoreReplOutput (unparseComponent goal)
+            <> makeKoreReplOutput (prettyComponent goal)
         , provenValue = makeAuxReplOutput "Reached bottom"
         }
   where
-    unparseComponent =
-        unparseToString . fmap hide . getRewritingPattern . transformation
+    prettyComponent =
+        renderDefault . pretty . fmap hide . getRewritingPattern
```

Comment:
This function should unparse a correct `kore` pattern, because we need to pipe it to `kast` in order to get the `K` representation.

- We could turn `Pattern` into `TermLike` here, and unparse that, but we would lose some debugging information (the separation between the term, predicate and substitution)
- We could keep the `Unparse` instance for `Conditional variable (TermLike variable)`, and turn the predicate into a term when unparsing.